action.yml

```diff
@@ -0,0 +1,155 @@
+      run: docker push ${{ env.DOCKER_IMAGE }}
+      shell: bash
+
+    - uses: dorny/paths-filter@v2
+      id: infrastructure-changes
+      with:
+        filters: |
+          terraform:
+            - 'terraform/**'
+
+    - name: deploy infrastructure
+      if: steps.infrastructure-changes.outputs.terraform == 'true'
+      uses: spring-media/1up-github-actions@deploy-infrastructure
+      with:
+        service-name: ${{ env.SERVICE_NAME }}
+        pkg-user: ${{ inputs.pkg-user }}
+        pkg-token: ${{ inputs.pkg-token }}
+        docker-image-tag: ${{ inputs.docker-image-tag }}
+        github-token: ${{ inputs.github-token }}
+        staging: ${{ inputs.staging }}
+
+    - name: Release - on staging
+      if: ${{ inputs.staging == 'true' && steps.infrastructure-changes.outputs.terraform == 'false' }}
+      uses: silinternational/ecs-deploy@master
+      with:
+        aws_access_key_cmd: --aws-access-key
+        aws_access_key: ${{ inputs.up-aws-access-key-id }}
+        aws_secret_key_cmd: --aws-secret-key
+        aws_secret_key: ${{ inputs.up-aws-secret-access-key }}
+        cluster_cmd: --cluster
+        cluster: up-cluster-staging
+        image_cmd: --image
+        image: ${{ env.DOCKER_IMAGE }}
+        region_cmd: --region
+        region: eu-west-1
+        service_name_cmd: --service-name
+        service_name: ${{ env.SERVICE_NAME }}
+        timeout_cmd: --timeout
+        timeout: 600
+
+    - name: Release - [on master]
+      if: github.ref == 'refs/heads/master' && steps.infrastructure-changes.outputs.terraform == 'false'
+      uses: silinternational/ecs-deploy@master
+      with:
+        aws_access_key_cmd: '--aws-access-key'
+        aws_access_key: ${{ inputs.up-aws-access-key-id }}
+        aws_secret_key_cmd: '--aws-secret-key'
+        aws_secret_key: ${{ inputs.up-aws-secret-access-key }}
+        cluster_cmd: '--cluster'
+        cluster: 'up-cluster-production'
+        image_cmd: '--image'
+        image: ${{ env.DOCKER_IMAGE }}
+        region_cmd: '--region'
+        region: 'eu-west-1'
+        service_name_cmd: '--service-name'
+        service_name: ${{ env.SERVICE_NAME }}
+        timeout_cmd: '--timeout'
+        timeout: '600'
+
+    - name: Report - setup [on master]
+      if: always() && github.ref == 'refs/heads/master'
+      run: |
+        GIT_COMMIT_MSG=$(git log -1 --pretty=format:'%s')
+        GIT_COMMIT_MSG_NO_SUFFIX="${GIT_COMMIT_MSG#*\(#}"
+        GIT_PR="${GIT_COMMIT_MSG_NO_SUFFIX%\)*}"
+        GIT_REPO_URL="https://github.com/${{ github.repository }}"
+        GIT_PR_URL="$GIT_REPO_URL/pull/$GIT_PR"
+        GIT_REPO_NAME="${{ github.repository }}"
+        echo "GIT_ACTIONS_URL=<$GIT_REPO_URL/actions/runs/${{ github.run_id }}|${GIT_REPO_NAME#*/}>" >> $GITHUB_ENV
+        [[ "$GIT_PR" =~ ^[0-9]+$ ]] && GIT_PR_MSG="<$GIT_PR_URL|$GIT_COMMIT_MSG>" || GIT_PR_MSG="$GIT_COMMIT_MSG"
+        echo "GIT_PR_MSG=$GIT_PR_MSG" >> $GITHUB_ENV
+      shell: bash
```

Comment:
Are you fine with moving shell: bash before the bash script? This applies to all steps using shell scripts :)

Reply:
what's the difference?

Reply:
Technically none, but in my regards a bit in terms of readability

Reply:
maybe difference in thinking, I feel at the end if more readable :D

Reply:
at least can find out easily where this is not added